### PR TITLE
feat(ingest): Add GenericAspectTransformer

### PR DESCRIPTION
**Pull request by amanda-her**
## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Comment by Khurzak** [comment]
With these changes, our aim is to create the infrastructure that will allow creating transformers to consume generic aspects.
A new GenericAspectTransformer with the following goals:
- Support GenericAspectClass instead of Aspect
- When an aspect is transformed return MetadataChangeProposalClass instead of MetadataChangeProposalWrapper

BaseTransformer has been extended to support MetadataChangeProposalClass

File: metadata-ingestion/src/datahub/ingestion/transformer/generic_aspect_transformer.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        custom aspects. They need to be upserted with MetadataChangeProposalClass instead of
+        MetadataChangeProposalWrapper used at the original method.
+        """
+        for envelope in record_envelopes:
+            if not self._should_process(envelope.record):
+                pass
+            elif isinstance(envelope.record, MetadataChangeEventClass):
+                self._record_mce(envelope.record)
+            elif isinstance(envelope.record, MetadataChangeProposalWrapper):
+                self._record_mcp(envelope.record)
```
Comment (asikowitz) [inline metadata-ingestion/src/datahub/ingestion/transformer/generic_aspect_transformer.py:80]: We can handle both these cases by:
- Pulling the appropriate aspect out of the MCE
- Converting the MCPW into a MCP with `make_mcp`

File: metadata-ingestion/src/datahub/ingestion/transformer/generic_aspect_transformer.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    """Transformer that does transform custom aspects using GenericAspectClass."""
+
+    def __init__(self):
+        super().__init__()
+
+    def transform_aspect(
+        self, entity_urn: str, aspect_name: str, aspect: Optional[Aspect]
+    ) -> Optional[Aspect]:
+        """Do not implement."""
+        pass
```
Comment (asikowitz) [inline metadata-ingestion/src/datahub/ingestion/transformer/generic_aspect_transformer.py:34]: I can see for simplicity why you inherited from `SimpleAspectTransformer`, but I think it's awkward we're essentially ignoring the main method of that class.

**Review by asikowitz — APPROVED** [review]
Thanks for your contribution to datahub! I think `GenericAspectTransformer` may be more appropriate as another mixin (like `LegacyMCETransformer` and `SingleAspectTransformer`, but this works as is.

**Pull request merged**